packages/grid/_modules_/grid/components/menu/GridMenu.tsx

```diff
@@ -39,7 +39,7 @@ const useStyles = makeStyles(
 export interface GridMenuProps extends Omit<PopperProps, 'onKeyDown'> {
   open: boolean;
   target: React.ReactNode;
-  onClickAway: (event?: React.MouseEvent<Document, MouseEvent>) => void;
+  onClickAway: (event: React.SyntheticEvent<Document, MouseEvent>) => void;
```

Comment:
This is not a synthetic event AFAIK. In v5, I see

```tsx
onClickAway: (event: MouseEvent | TouchEvent) => void;
```

in the source of ClickAwayListener. So I guess?
```suggestion
  onClickAway: (event: MouseEvent | TouchEvent) => void;
```

Reply:
Yes but in v4 (which we are still using) is `onClickAway: (event: React.MouseEvent<Document>) => void;`

https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/ClickAwayListener/ClickAwayListener.d.ts#L20

For now I can type it to satisfy the linter and later when we fully migrate to v5 fix the typing (it will inevitably fail on the typescript step so there is no way we will miss it).

Reply:
Ok, sounds fair